torchao/quantization/quant_api.py

```diff
@@ -665,9 +666,12 @@ def apply_int4_weight_only_quant(weight):
         quant_min = 0
         quant_max = 15
         eps = 1e-6
-        preserve_zero = False
+        preserve_zero = zero_point_domain == ZeroPointDomain.INT
         zero_point_dtype = torch.bfloat16
-        zero_point_domain = ZeroPointDomain.FLOAT
+
+        if isinstance(layout, TensorCoreTiledLayout):
```

Comment:
I think zero_point_domain should probably default to None, which means we'll set it based on layout, but when it is not None, we'd need to check it for TensorCoreTiledLayout, MarlinSparseLayout and IntCPULayout

Reply:
fixed in https://github.com/pytorch/ao/pull/1401/commits/1dbf9aa7bf4b1c0d4aba686e598d4264ed03ac64